packages/blade/src/components/Button/BaseButton/__tests__/BaseButton.web.test.tsx

```diff
@@ -123,4 +122,228 @@ describe('<BaseButton />', () => {
     fireEvent.click(button);
     expect(onClick).toHaveBeenCalledTimes(1);
   });
+
+  it('should render secondary variant button', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(
+      <BaseButton variant="secondary">{buttonText}</BaseButton>,
+    );
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render disabled secondary variant button', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(
+      <BaseButton variant="secondary" isDisabled={true}>
+        {buttonText}
+      </BaseButton>,
+    );
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render tertiary variant button', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(<BaseButton variant="tertiary">{buttonText}</BaseButton>);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render disabled tertiary variant button', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(
+      <BaseButton variant="tertiary" isDisabled={true}>
+        {buttonText}
+      </BaseButton>,
+    );
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render positive intent low contrast button', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(
+      <BaseButton intent="positive" contrast="low">
+        {buttonText}
+      </BaseButton>,
+    );
+    expect(container).toMatchSnapshot();
+  });
+  it('should render disabled positive intent low contrast button', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(
+      <BaseButton intent="positive" contrast="low" isDisabled={true}>
+        {buttonText}
+      </BaseButton>,
+    );
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render negative intent low contrast button', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(
+      <BaseButton intent="negative" contrast="low">
+        {buttonText}
+      </BaseButton>,
+    );
+    expect(container).toMatchSnapshot();
+  });
+  it('should render disabled negative intent low contrast button', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(
+      <BaseButton intent="negative" contrast="low" isDisabled={true}>
+        {buttonText}
+      </BaseButton>,
+    );
```

Comment:
tests for checking a11y? using that axe tool with no violations and testing all the roles and a11y labels?